### PR TITLE
Feat add like functionality

### LINKED ISSUE #9: Feat Add Like Functionality
Add the videos liked to the page and manage the state accordingly.

### PR DESCRIPTION
**Pull request by vaibhav-mahalle**
- Added Like and dislike functionality
- Added watch later functionality.
- Added liked videos page

### CODE REVIEW

File: src/Pages/Liked/Liked.css [hunk 1]
```diff
@@ -0,0 +1,10 @@
+.msg-container{
+    width: 30rem;
+    height: 5rem;
+    /* background-color: var(--vl-primary); */
```
Comment (nakulsharma15) [inline src/Pages/Liked/Liked.css]: You can remove commented CSS if it's no longer needed.

File: src/Pages/Liked/Liked.css [hunk 2]
```diff
@@ -0,0 +1,10 @@
...
+    display: flex;
+    align-items: center;
+    justify-content: center;
```
Comment (nakulsharma15) [inline src/Pages/Liked/Liked.css:6]: You can use a utility class for it.

File: src/context/Video/getVideos.js [hunk 3]
```diff
@@ -0,0 +1,11 @@
+import axios from "axios";
+export const getAllVideos = async () => {
+    try {
+      const res = await axios.get("/api/videos");
+      if (res.status === 200) {
+        videoDispatch({type: "SET_VIDEOS", payload: res.data.videos});
+      }
+    } catch(error){
+      console.error("Error in setting videos", error);
```
Comment (nakulsharma15) [inline src/context/Video/getVideos.js]: You can add a toast here to show error to user.

File: src/components/VideoItem/VideoItem.jsx [hunk 4]
```diff
@@ -2,25 +2,34 @@ import React, { useState } from "react";
...
+  const [watchlater, setWatchLater] = useState([]);
 
+  const addToLikedVideos = () => {
+    setLikedVideos([...likedVideos, item]); 
+  }
+  const addToWatchLater = () => {
+    setWatchLater([...watchlater, item]); 
+  }
+  console.log(item);
+  console.log(likedVideos);
```
Comment (nakulsharma15) [inline src/components/VideoItem/VideoItem.jsx]: You can remove console.log after it is no longer needed.

File: src/context/Video/history.js [hunk 5]
```diff
@@ -0,0 +1,53 @@
...
+    try{
+        const res = await axios.delete('/api/user/history/', {headers:{authorization:isLoggedIn}});
+        if(res.status === 200){
+            videoDispatch({type: "REMOVE_FROM_HISTORY", payload: res.data.history});
+            ToastMsg("Video Removed from history","success")
+        }
+        
+    }
+    catch(error){
+
```
Comment (nakulsharma15) [inline src/context/Video/history.js:26]: You can add a toast here to show error message to the user.

File: src/Pages/VideoPage/index.js [hunk 6]
```diff
@@ -1,35 +1,59 @@
...
       autoplay: 0,
     },
   };
+
+  const findRequiredVideo = (videosData) => videosData.find(item => item.videoId === videoId);
+
+  const videoData = findRequiredVideo(videos);
+  const isLiked = isVideoLiked(videoData?.videoId, liked);
+
+  console.log(videoData)
```
Comment (nakulsharma15) [inline src/Pages/VideoPage/index.js]: Remove console.log after it's not needed.

**Review by nakulsharma15 — COMMENTED** [review]
Suggested some changes.  Apart from that ,code looks nice and clean. I liked how you named the css classes and I also liked the overall naming of the functions and variables.

File: src/Pages/VideoPage/index.js [hunk 7]
```diff
@@ -1,35 +1,59 @@
...
 import { useParams } from "react-router-dom";
 import YouTube from "react-youtube";
-import { useVideo } from "../../context/Video/context";
-import "./VideoPage.css";
 import { BiLike } from "react-icons/bi";
-import { MdPlaylistAdd, MdHistory } from "react-icons/md";
+import { MdPlaylistAdd, MdOutlineWatchLater } from "react-icons/md";
+import { LikeVideo, unLikeVideo, isVideoLiked} from "../../context/Video/liked";
+import { useVideo } from "../../context/Video/context";
+import { useAuth } from "../../context/Auth/context";
```
Comment (rahulcs754) [inline src/Pages/VideoPage/index.js]: you can create an index file and export all context and you have to write only one line then

File: src/context/Video/history.js [hunk 8]
```diff
@@ -0,0 +1,53 @@
...
+const deleteAllFromHistory = async () => {
+    try{
+        const res = await axios.delete('/api/user/history/all', {headers: {authorization: isLoggedIn}});
+        if(res.status === 200){
+            videoDispatch({type: "REMOVE_ALL_FROM_HISTORY", payload: res.data.history});
+            ToastMsg("All videos removed from history","success")
+        }
+    }
+    catch(error){
+
```
Comment (rahulcs754) [inline src/context/Video/history.js:39]: i think you have provide toast for error

**Review by rahulcs754 — APPROVED** [review]
The code looks good and clean. just small suggestion

File: src/Pages/Liked/Liked.css (hunk 1 above)
Comment (vaibhav-mahalle) [reply src/Pages/Liked/Liked.css]: thanks for pointing it out, I have made the required changes.

File: src/context/Video/getVideos.js (hunk 3 above)
Comment (vaibhav-mahalle) [reply src/context/Video/getVideos.js]: thanks for pointing it out, I have made the required changes.

File: src/context/Video/history.js (hunk 5 above)
Comment (vaibhav-mahalle) [reply src/context/Video/history.js:26]: thanks for pointing it out, I have made the required changes.

File: src/Pages/VideoPage/index.js (hunk 7 above)
Comment (vaibhav-mahalle) [reply src/Pages/VideoPage/index.js]: thanks for pointing it out, I have made the required changes.

File: src/context/Video/history.js (hunk 8 above)
Comment (vaibhav-mahalle) [reply src/context/Video/history.js:39]: thanks for pointing it out, I have made the required changes.

**Pull request merged**